InformationScripting/src/queries/CompositeQuery.cpp

```diff
@@ -76,7 +77,13 @@ QList<TupleSet> CompositeQuery::execute(QList<TupleSet> input)
 					return m.outputFrom_ == currentNode && m.outputIndex_ == outIndex;
 				});
 				Q_ASSERT(inputIt != receiver->inputMap_.end());
-				auto output = hasOutput ? currentNode->calculatedOutputs_[outIndex] : TupleSet();
+				TupleSet output;
+				if (hasOutput)
+				{
+					// early abort in case of error:
+					if (currentNode->calculatedOutputs_[outIndex]) output = currentNode->calculatedOutputs_[outIndex].value();
+					else return { currentNode->calculatedOutputs_[outIndex].error() };
```

Comment:
Well, what about a hypothetical scenario where two independent branches of a composite query both have an error. Should we just pick one of those at random and return it? Especially later, if we allow parallel execution at some point, this will be a problem.

I guess if you convert everything to `Optional<QList<..>>` then in the sequential case, we can still get away with one error.

Might be nicer/easier though to treat errors and warnings almost identically in `Optional`, so to have a list that can hold either.